### PR TITLE
Issue 2991147   DD-418 update taboola JS encode json correct way

### PR DESCRIPTION
**Pull request by jon-edery**

### CODE REVIEW

**Comment by jon-edery** [comment]
Once this PR has been reviewed and tested the patch needs to be generated and posted on Drupal.org  project ->  https://www.drupal.org/project/taboola/issues/2991147.

A new release for the module will need to be created.

The make file on the evo site will need to point to the new release.

**Review by marcelovani — CHANGES_REQUESTED** [review]
I think we should commit this against 7.x-3.x

File: taboola.module [hunk 1]
```diff
@@ -288,10 +289,16 @@ function taboola_get_widget_setting_js($block_index) {
...
   // Get target type setting.
   $target_type = taboola_get_block_setting($block_index, 'target_type');
-  $target_type_json = !empty($target_type) ? ", target_type:'" . check_plain($target_type) . "'" : '';
+
+  $widget_settings = array();
+  $widget_settings['mode'] = $mode;
+  $widget_settings['container'] = $container;
+  $widget_settings['placement'] = $placement;
+  $widget_settings['target_type'] = !empty($target_type) ?  $target_type : '';
+  $widget_settings_json = json_encode($widget_settings);
```
Comment (davidgrayston) [inline taboola.module]: Perhaps we should use drupal_json_encode() here?

**Review by davidgrayston — CHANGES_REQUESTED** [review]

**Comment by davidgrayston** [comment]
@marcelovani depends if this is a breaking change or is backward compatible - looks like it might be ok on 2.x - @jon-edery can we confirm with product/Taboola?

File: taboola.module [hunk 2]
```diff
@@ -79,7 +79,8 @@ function taboola_page_alter(&$page) {
       "window._taboola = window._taboola || [];",
       "_taboola.push({" . $loader_identifier . ":'auto'});",
```
Comment (davidgrayston) [inline taboola.module]: @jon-edery can we also `drupal_json_encode()` this?

File: taboola.module (hunk 1 above)
Comment (davidgrayston) [reply taboola.module]: ... it adds `JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT` options

File: taboola.module [hunk 3]
```diff
@@ -79,7 +79,8 @@ function taboola_page_alter(&$page) {
...
       "!function (e, f, u) { e.async = 1; e.src = u; f.parentNode.insertBefore(e, f);}",
```
Comment (davidgrayston) [inline taboola.module]: @jon-edery looks like it's missing the 4th `i` argument and JS that uses is:
```
    if (!document.getElementById(i)){
      e.async = 1;
      e.src = u;
      e.id = i;
      f.parentNode.insertBefore(e, f);
    }
```

**Comment by jon-edery** [comment]
requested updates added.

**Review by jimwills123 — APPROVED** [review]
all changes made

**Pull request merged**